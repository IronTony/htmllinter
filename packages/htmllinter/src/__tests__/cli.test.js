import { resolve } from 'path';
import execa from 'execa';

const { sync: spSync } = execa;
const cliPath = resolve(__dirname, '../../bin/htmllinter.js');

describe('Testing CLI behavior', () => {
  describe('input pattern', () => {
    it('should show help when no arg is passed with proper err message', () => {
      const result = spSync(cliPath);
      expect(result.exitCode).toBe(0);
      expect(result.stderr).toBe(
        '[HTMLLINTER]: no argument passed. Please provide the input file(s) atleast'
      );
      expect(result.stdout).toMatchSnapshot('help');
    });

    it('should show help when when non-existing input is passed with proper err message', () => {
      const inputFileName = 'nonExist.html';
      const result = spSync(cliPath, [inputFileName]);
      expect(result.stderr).toBe(
        `[HTMLLINTER]: no file found with input pattern : ${inputFileName}`
      );
      expect(result.exitCode).toBe(0);
      expect(result.stdout).toMatchSnapshot('help');
    });

    it('should show message when wrong pattern is pass', () => {
      const pattern = 'dir,dir2';
      const result = spSync(cliPath, [pattern]);
      expect(result.stderr).toBe(
        `[HTMLLINTER]: no file found with input pattern : ${pattern}`
      );
      expect(result.exitCode).toBe(0);
    });
  });

  describe('Config flag', () => {
    it('should show message when correct path to config is passed but not present', () => {
      const configFilePath = './packages';
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig/*.html';
      const result = spSync(cliPath, [pattern, '--config', configFilePath], {
        reject: false,
      });

      const errArr = result.stderr.split(' ');
      expect(errArr).toContain('Cannot');
      expect(errArr).toContain('find');
      expect(errArr).toContain('module');
      expect(result.stdout).toBe('');
      expect(result.exitCode).toBe(1);
    });

    it('should show message when htmllinter.config.js is passed along with the path even if correct', () => {
      const configFilePath =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig/htmllinter.config.js';
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig/*.html';
      const result = spSync(cliPath, [pattern, '--config', configFilePath], {
        reject: false,
      });

      const errArr = result.stderr.split(' ');
      expect(errArr).toContain('Cannot');
      expect(errArr).toContain('find');
      expect(errArr).toContain('module');
      expect(result.stdout).toBe('');
      expect(result.exitCode).toBe(1);
    });
  });

  describe('default config : fixture : defaultConfig', () => {
    it('should show the linting errors  (no htmllinter.config.js present) ', () => {
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/defaultConfig/input.html';
      const result = spSync(cliPath, [pattern], {
        reject: false,
        stdio: 'pipe',
      });
      expect(result.exitCode).toBe(1);
      expect(result.stdout).toMatchSnapshot('default config');
    });

    it('should not show the linting errors  (no htmllinter.config.js present)  : no-error', () => {
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/defaultConfig/noErr.html';
      const result = spSync(cliPath, [pattern], {
        reject: false,
        stdio: 'pipe',
      });

      expect(result.exitCode).toBe(0);
      expect(result.stdout).toMatchSnapshot('default config no error');
    });
  });

  describe('extend config : fixture : extendConfig', () => {
    it('should show the linting errors  ( htmllinter.config.js present) ', () => {
      const configFilePath =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig';
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig/input.html';
      const result = spSync(cliPath, [pattern, '-c', configFilePath], {
        reject: false,
        stdio: 'pipe',
      });

      expect(result.exitCode).toBe(1);
      expect(result.stdout).toMatchSnapshot('extend config');
    });

    it('should not show the linting errors  ( htmllinter.config.js present) as non-Error files ', () => {
      const configFilePath =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig';
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/extendConfig/noError.html';
      const result = spSync(cliPath, [pattern, '-c', configFilePath], {
        reject: false,
        stdio: 'pipe',
      });

      expect(result.exitCode).toBe(0);
      expect(result.stdout).toMatchSnapshot('extend config');
    });
  });

  describe('extend config with rule off : fixture : extendConfigRuleOff', () => {
    it('should show the linting errors  ( htmllinter.config.js present) ', () => {
      const configFilePath =
        './packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff';
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/input.html';
      const result = spSync(cliPath, [pattern, '-c', configFilePath], {
        reject: false,
        stdio: 'pipe',
      });

      expect(result.exitCode).toBe(1);
      expect(result.stdout).toMatchSnapshot('extend config rule off');
    });

    it('should not show the linting errors  ( htmllinter.config.js present) as non-Error files ', () => {
      const configFilePath =
        './packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff';
      const pattern =
        './packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/noError.html';
      const result = spSync(cliPath, [pattern, '-c', configFilePath], {
        reject: false,
        stdio: 'pipe',
      });
      expect(result.exitCode).toBe(0);
      expect(result.stdout).toMatchSnapshot('extend config rule off');
    });
  });
});
