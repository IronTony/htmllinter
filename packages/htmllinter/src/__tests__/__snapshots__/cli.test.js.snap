// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CLI behavior default config : fixture : defaultConfig should not show the linting errors  (no htmllinter.config.js present)  : no-error: default config no error 1`] = `
"[HTMLLINTER] : Checking ./packages/htmllinter/src/__tests__/fixtures/defaultConfig/noErr.html 


[HTMLLINTER] : 0 errors found for file ./packages/htmllinter/src/__tests__/fixtures/defaultConfig/noErr.html

 Output for filename : ./packages/htmllinter/src/__tests__/fixtures/defaultConfig/noErr.html 

[90m┌─────────[39m[90m┬───────────┐[39m
[90m│[39m[31m Message [39m[90m│[39m[31m Rule Name [39m[90m│[39m
[90m└─────────[39m[90m┴───────────┘[39m"
`;

exports[`Testing CLI behavior default config : fixture : defaultConfig should show the linting errors  (no htmllinter.config.js present) : default config 1`] = `
"[HTMLLINTER] : Checking ./packages/htmllinter/src/__tests__/fixtures/defaultConfig/input.html 

 Output for filename : ./packages/htmllinter/src/__tests__/fixtures/defaultConfig/input.html 

[90m┌──────────────────────────────────────────[39m[90m┬─────────────────┐[39m
[90m│[39m[31m Message                                  [39m[90m│[39m[31m Rule Name       [39m[90m│[39m
[90m├──────────────────────────────────────────[39m[90m┼─────────────────┤[39m
[90m│[39m the tag < h3 > has no content.           [90m│[39m no-empty-tag    [90m│[39m
[90m├──────────────────────────────────────────[39m[90m┼─────────────────┤[39m
[90m│[39m [HTMLINTER] duplicate ids asd adsasd @ a [90m│[39m no-duplicate-id [90m│[39m
[90m└──────────────────────────────────────────[39m[90m┴─────────────────┘[39m"
`;

exports[`Testing CLI behavior extend config : fixture : extendConfig should not show the linting errors  ( htmllinter.config.js present) as non-Error files : extend config 1`] = `
"[HTMLLINTER] : Checking ./packages/htmllinter/src/__tests__/fixtures/extendConfig/noError.html 


[HTMLLINTER] : 0 errors found for file ./packages/htmllinter/src/__tests__/fixtures/extendConfig/noError.html

 Output for filename : ./packages/htmllinter/src/__tests__/fixtures/extendConfig/noError.html 

[90m┌─────────[39m[90m┬───────────┐[39m
[90m│[39m[31m Message [39m[90m│[39m[31m Rule Name [39m[90m│[39m
[90m└─────────[39m[90m┴───────────┘[39m"
`;

exports[`Testing CLI behavior extend config : fixture : extendConfig should show the linting errors  ( htmllinter.config.js present) : extend config 1`] = `
"[HTMLLINTER] : Checking ./packages/htmllinter/src/__tests__/fixtures/extendConfig/input.html 

 Output for filename : ./packages/htmllinter/src/__tests__/fixtures/extendConfig/input.html 

[90m┌───────────────────────────────────────────────────────────────────────────────────────────────────────[39m[90m┬──────────────────────────────┐[39m
[90m│[39m[31m Message                                                                                               [39m[90m│[39m[31m Rule Name                    [39m[90m│[39m
[90m├───────────────────────────────────────────────────────────────────────────────────────────────────────[39m[90m┼──────────────────────────────┤[39m
[90m│[39m the tag < p > has no content.                                                                         [90m│[39m no-empty-tag                 [90m│[39m
[90m├───────────────────────────────────────────────────────────────────────────────────────────────────────[39m[90m┼──────────────────────────────┤[39m
[90m│[39m the tag < h3 > has no content.                                                                        [90m│[39m no-empty-tag                 [90m│[39m
[90m├───────────────────────────────────────────────────────────────────────────────────────────────────────[39m[90m┼──────────────────────────────┤[39m
[90m│[39m [HTMLINTER] duplicate ids asd adsasd @ a                                                              [90m│[39m no-duplicate-id              [90m│[39m
[90m├───────────────────────────────────────────────────────────────────────────────────────────────────────[39m[90m┼──────────────────────────────┤[39m
[90m│[39m the attribute \\"enable\\" seems to be boolean with value \\"true\\",please conside using \\"<p enable>...</p>\\" [90m│[39m no-bool-true-explicit-define [90m│[39m
[90m│[39m                                                                                                       [90m│[39m                              [90m│[39m
[90m└───────────────────────────────────────────────────────────────────────────────────────────────────────[39m[90m┴──────────────────────────────┘[39m"
`;

exports[`Testing CLI behavior extend config with rule off : fixture : extendConfigRuleOff should not show the linting errors  ( htmllinter.config.js present) as non-Error files : extend config rule off 1`] = `
"[HTMLLINTER] : Checking ./packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/noError.html 


[HTMLLINTER] : 0 errors found for file ./packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/noError.html

 Output for filename : ./packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/noError.html 

[90m┌─────────[39m[90m┬───────────┐[39m
[90m│[39m[31m Message [39m[90m│[39m[31m Rule Name [39m[90m│[39m
[90m└─────────[39m[90m┴───────────┘[39m"
`;

exports[`Testing CLI behavior extend config with rule off : fixture : extendConfigRuleOff should show the linting errors  ( htmllinter.config.js present) : extend config rule off 1`] = `
"[HTMLLINTER] : Checking ./packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/input.html 

 Output for filename : ./packages/htmllinter/src/__tests__/fixtures/extendConfigRuleOff/input.html 

[90m┌──────────────────────────────────────────[39m[90m┬─────────────────┐[39m
[90m│[39m[31m Message                                  [39m[90m│[39m[31m Rule Name       [39m[90m│[39m
[90m├──────────────────────────────────────────[39m[90m┼─────────────────┤[39m
[90m│[39m [HTMLINTER] duplicate ids asd adsasd @ a [90m│[39m no-duplicate-id [90m│[39m
[90m└──────────────────────────────────────────[39m[90m┴─────────────────┘[39m"
`;

exports[`Testing CLI behavior input pattern should show help when no arg is passed with proper err message: help 1`] = `
"
Usage :

$ htmllinter <input> [-c|--config]

input           - it is the input(html) file name, it accepts glob pattern.
                example :
                  - \\"input.html\\"
                  - \\"input\\"
                  - \\"/*.html\\"
                  - \\".\\"

Options
  -c, --config    - path to htmllinter.config.js.
                    default is \\".\\"
                    example :
                    - '.'
                    - './dir/'

Example

    $ htmllinter input.html
    $ htmllinter input -c ../
    $ htmllinter **/*"
`;

exports[`Testing CLI behavior input pattern should show help when when non-existing input is passed with proper err message: help 1`] = `
"
Usage :

$ htmllinter <input> [-c|--config]

input           - it is the input(html) file name, it accepts glob pattern.
                example :
                  - \\"input.html\\"
                  - \\"input\\"
                  - \\"/*.html\\"
                  - \\".\\"

Options
  -c, --config    - path to htmllinter.config.js.
                    default is \\".\\"
                    example :
                    - '.'
                    - './dir/'

Example

    $ htmllinter input.html
    $ htmllinter input -c ../
    $ htmllinter **/*"
`;
